hydromt/models/model_api.py

```diff
@@ -1273,6 +1281,8 @@ def write_geoms(self, fn: str = "geoms/{name}.geojson", **kwargs) -> None:
             _fn = join(self.root, fn.format(name=name))
             if not isdir(dirname(_fn)):
                 os.makedirs(dirname(_fn))
+            if kwargs.get("driver") == "GeoJSON" and to_wgs84:
+                gdf.to_crs(4326, inplace=True)
```

Comment:
Avoid that the original gdf in self.geoms get modified
```suggestion
                gdf = gdf.to_crs(4326)
```